### PR TITLE
feat(1830): build cache to disk

### PR DESCRIPTION
**Pull request by parthasl**
## Context

All builds in the pipeline should be automatically sent to the build cluster assigned at pipeline creation time

## Objective

Assign buildCluster for pipelines and builds

## References

[1830](https://github.com/screwdriver-cd/screwdriver/issues/1830)

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: lib/pipeline.js [hunk 1]
```diff
@@ -1187,10 +1189,34 @@ class PipelineModel extends BaseModel {
...
+                    });
+
+                    if (buildClusterName) {
+                        if (!this.annotations) {
+                            this.annotations = {};
+                        }
+                        this.annotations[buildClusterAnnotation] = buildClusterName;
+                    }
+                } /* else {
+                    /* eslint no-lonely-if:
```
Comment (jithine) [inline lib/pipeline.js]: remove commented code.

File: lib/pipelineFactory.js [hunk 2]
```diff
@@ -63,10 +67,35 @@ class PipelineFactory extends BaseFactory {
...
+                    });
+
+                    if (buildClusterName) {
+                        if (!modelConfig.annotations) {
+                            modelConfig.annotations = {};
+                        }
+                        modelConfig.annotations[buildClusterAnnotation] = buildClusterName;
+                    }
+                } /* else {
+                    /* eslint no-lonely-if:
```
Comment (jithine) [inline lib/pipelineFactory.js]: remove commented code

**Review by jithine — CHANGES_REQUESTED** [review]
Looking good. Can you remove commented out code ?

File: lib/pipeline.js (hunk 1 above)
Comment (parthasl) [reply lib/pipeline.js]: done

File: lib/pipelineFactory.js (hunk 2 above)
Comment (parthasl) [reply lib/pipelineFactory.js]: done

File: test/lib/pipeline.test.js [hunk 3]
```diff
@@ -1837,6 +1868,301 @@ describe('Pipeline Model', () => {
             });
         });
 
+        /* it('throw err - multipleBuildClusterDisabled with cluster annotations - ' +
```
Comment (jithine) [inline test/lib/pipeline.test.js]: can you remove this as well , if not needed ?

File: test/lib/pipelineFactory.test.js [hunk 4]
```diff
@@ -150,6 +179,350 @@ describe('Pipeline Factory', () => {
...
+                admins,
+                annotations: { 'screwdriver.cd/prChain': 'fork' }
+            }).then((model) => {
+                assert.calledWith(scm.decorateUrl, { scmUri, scmContext, token: 'foo' });
+                assert.calledWith(datastore.save, saveConfig);
+                assert.instanceOf(model, Pipeline);
+            });
+        });
+
+        /* it('throw err - multipleBuildClusterDisabled with cluster annotations - ' +
```
Comment (jithine) [inline test/lib/pipelineFactory.test.js]: can you remove this as well , if not needed ?

File: test/lib/pipeline.test.js (hunk 3 above)
Comment (parthasl) [reply test/lib/pipeline.test.js]: done

File: test/lib/pipelineFactory.test.js (hunk 4 above)
Comment (parthasl) [reply test/lib/pipelineFactory.test.js]: done

**Review by jithine — APPROVED** [review]

**Pull request merged**